lib/options-handler.js

```diff
@@ -131,22 +131,30 @@ export class ClientOptionsHandler {
                     const toolBaseName = `tools.${tool}`;
                     const className = this.compilerProps(lang, toolBaseName + '.class');
                     const Tool = getToolTypeByKey(className);
-                    tools[lang][tool] = new Tool({
-                        id: tool,
-                        name: this.compilerProps(lang, toolBaseName + '.name'),
-                        type: this.compilerProps(lang, toolBaseName + '.type'),
-                        exe: this.compilerProps(lang, toolBaseName + '.exe'),
-                        exclude: this.compilerProps(lang, toolBaseName + '.exclude'),
-                        includeKey: this.compilerProps(lang, toolBaseName +'.includeKey'),
-                        options: this.compilerProps(lang, toolBaseName + '.options'),
-                        languageId: this.compilerProps(lang, toolBaseName + '.languageId'),
-                        stdinHint: this.compilerProps(lang, toolBaseName + '.stdinHint'),
-                        compilerLanguage: lang,
-                    },
-                    {
-                        ceProps: this.ceProps,
-                        compilerProps: (propname) => this.compilerProps(lang, propname),
-                    });
+
+                    const toolPath = this.compilerProps(lang, toolBaseName + '.exe');
+                    try {
+                        const res = fs.statSync(toolPath);
```

Comment:
I think using fs.existsSync() would be more readable here (and then use if() instead of catching an exception)